sx13xx-conf/src/global_conf.rs

```diff
@@ -0,0 +1,327 @@
+        }
+        if let Some(frequency) = self.chan_multiSF_5.frequency(&self.radio_0, &self.radio_1) {
+            frequencies.push(frequency);
+        }
+        if let Some(frequency) = self.chan_multiSF_6.frequency(&self.radio_0, &self.radio_1) {
+            frequencies.push(frequency);
+        }
+        if let Some(frequency) = self.chan_multiSF_7.frequency(&self.radio_0, &self.radio_1) {
+            frequencies.push(frequency);
+        }
+
+        // iterate through all frequencies and confirm that they are between
+        // tx_freq_min and tx_freq_max
+        let mut valid_tx = true;
+        if let (Some(lb), Some(ub)) = (self.radio_0.tx_freq_min, self.radio_0.tx_freq_max) {
+            for frequency in frequencies {
+                if frequency > ub || frequency < lb {
+                    valid_tx = false;
+                }
+            }
+        } else {
+            panic!("No tx freq max and min for radio_0!")
+        }
+
+        // prepare the summary to be printed
+        let mut summary = String::new();
+        summary.push_str("1        ");
+        summary.push_str(&self.chan_multiSF_0.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n2        ");
+        summary.push_str(&self.chan_multiSF_1.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n3        ");
+        summary.push_str(&self.chan_multiSF_2.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n4        ");
+        summary.push_str(&self.chan_multiSF_3.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n5        ");
+        summary.push_str(&self.chan_multiSF_4.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n6        ");
+        summary.push_str(&self.chan_multiSF_5.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n7        ");
+        summary.push_str(&self.chan_multiSF_6.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\n8        ");
+        summary.push_str(&self.chan_multiSF_7.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\nFat LoRa ");
+        summary.push_str(&self.chan_Lora_std.summary(&self.radio_0, &self.radio_1));
+        summary.push_str("\nFSK      ");
+        summary.push_str(&self.chan_FSK.summary(&self.radio_0, &self.radio_1));
+        if !valid_tx {
+            summary.push_str("\nWARNING: Cannot transmit on all uplink frequencies for POC!");
+        }
+        summary
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct Radio {
+    freq: isize,
+    tx_freq_min: Option<isize>,
+    tx_freq_max: Option<isize>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct Channel {
+    enable: bool,
+    #[serde(flatten)]
+    config: Option<ChannelEnabled>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct ChannelEnabled {
+    r#if: isize,
+    radio: usize,
+}
+
+impl Channel {
+    fn frequency(&self, radio_0: &Radio, radio_1: &Radio) -> Option<isize> {
```

Comment:
I'm not a huge fan of this nesting. An alternative could be:

```rust
    fn frequency(&self, radio_0: &Radio, radio_1: &Radio) -> Option<isize> {
        if !self.enable {
            return None;
        }
        let &ChannelEnabled { r#if, radio } = self
            .config
            .as_ref()
            .expect("LoRa Channel enabled but no 'radio' and/or no 'if'");
        Some(match radio {
            0 => radio_0.freq + r#if,
            1 => radio_1.freq + r#if,
            _ => panic!("invalid radio!"),
        })
    }
```

Reply:
Thanks. Did this